src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -1351,6 +1362,11 @@ public DType Add(DName name, DType type)
             Contracts.Assert(name.IsValid);
             type.AssertValid();
 
+            if (IsSealed)
+            {
+                throw new InvalidOperationException();
```

Comment:
If we throw an exception, can this break something? Would returning an error type be safer?

Reply:
Removed.